browser/ui/views/overlay/brave_video_overlay_window_views.cc

```diff
@@ -18,20 +25,173 @@
+      canvas->DrawCircle(
+          gfx::Point(line_bounds.right(), line_bounds.CenterPoint().y() - 1),
+          thumb_animation_.is_animating()
+              ? thumb_animation_.CurrentValueBetween(kLineHeight / 2,
+                                                     kThumbRadius)
+              : kThumbRadius,
+          flags);
+    }
+  }
+
+  void OnMouseEntered(const ui::MouseEvent& event) override {
+    Slider::OnMouseEntered(event);
+    mouse_hovered_ = true;
+    thumb_animation_.Show();
+  }
+
+  void OnMouseExited(const ui::MouseEvent& event) override {
+    Slider::OnMouseExited(event);
+    mouse_hovered_ = false;
+    if (!ShouldShowThumb()) {
+      thumb_animation_.Hide();
+    }
+  }
+
+  void AnimationProgressed(const gfx::Animation* animation) override {
+    if (animation == &thumb_animation_) {
+      SchedulePaint();
+      return;
+    }
+
+    Slider::AnimationProgressed(animation);
+  }
+
+  void AnimationEnded(const gfx::Animation* animation) override {
+    if (animation == &thumb_animation_) {
+      return;
+    }
+    Slider::AnimationEnded(animation);
+  }
+
+  // views::SliderListener:
+  void SliderValueChanged(views::Slider* sender,
+                          float value,
+                          float old_value,
+                          views::SliderChangeReason reason) override {
+    listener_->SliderValueChanged(sender, value, old_value, reason);
+  }
+
+  void SliderDragStarted(views::Slider* sender) override {
+    dragging_ = true;
+    listener_->SliderDragStarted(sender);
+  }
+
+  void SliderDragEnded(views::Slider* sender) override {
+    dragging_ = false;
+    if (!ShouldShowThumb()) {
+      thumb_animation_.Hide();
+    }
+    listener_->SliderDragEnded(sender);
+  }
+
+ private:
+  bool ShouldShowThumb() const { return dragging_ || mouse_hovered_; }
+
+  raw_ptr<views::SliderListener> listener_;
+
+  bool dragging_ = false;
+  bool mouse_hovered_ = false;
+
+  gfx::SlideAnimation thumb_animation_{this};
+};
+
 }  // namespace
 
 BraveVideoOverlayWindowViews::BraveVideoOverlayWindowViews(
     content::VideoPictureInPictureWindowController* controller)
     : VideoOverlayWindowViews(controller) {}
 
+BraveVideoOverlayWindowViews::~BraveVideoOverlayWindowViews() = default;
+
 void BraveVideoOverlayWindowViews::SetUpViews() {
   VideoOverlayWindowViews::SetUpViews();
   UpdateControlIcons();
+
+  timestamp_ =
+      controls_container_view_->AddChildView(std::make_unique<views::Label>());
+  timestamp_->SetEnabledColorId(kColorPipWindowForeground);
+  timestamp_->SetSubpixelRenderingEnabled(false);
+  timestamp_->SetAutoColorReadabilityEnabled(false);
+  timestamp_->SetElideBehavior(gfx::NO_ELIDE);
+  timestamp_->SetPaintToLayer(ui::LAYER_TEXTURED);
+  timestamp_->layer()->SetFillsBoundsOpaquely(false);
+  timestamp_->layer()->SetName("Timestamp");
+
+  // Unlike other controls, seeker should be visible even when mouse is not
```

Comment:
IMO, it would be better to show progress bar only when hovered. We decided to show it always?

Reply:
@rebron @aguscruiz , What do you think about this?